### PR TITLE
add update test

### PR DESCRIPTION
**Pull request by amityadav0**
## Proposed Changes

+ Add update test
+ fix typo

### CODE REVIEW

File: packages/algob/test/lib/tx.ts [hunk 1]
```diff
@@ -668,3 +668,53 @@ describe("Deploy, Delete transactions test in run mode", () => {
...
+      fromAccount: bobAcc,
+      appID: appInfo["application-index"],
+      newApprovalProgram: "approval.teal",
+      newClearProgram: "clear.teal",
+      payFlags: {}
+    };
+
+    await executeTransaction(deployer, execParams);
+  });
+});
```
Comment (ratik21) [inline packages/algob/test/lib/tx.ts:720]: Can we add one/two more tests here:
+ `deploySSC` in deploy mode (ie. cp is created), then do `updateSSC` and verify checkpoint(`deployer.getApp`) is same as previous one i.e checkpoint is not updated.

+ `deploySSC` in *RUN* mode. Try to update SSC in deploy mode (not sure what happens here, is the transaction rejected?) 

What do you think?

Comment (amityadav0) [reply packages/algob/test/lib/tx.ts:720]: yes, will be good to have these.

File: packages/algob/test/lib/tx.ts [hunk 2]
```diff
@@ -668,3 +668,53 @@ describe("Deploy, Delete transactions test in run mode", () => {
...
+      localInts: 1,
+      localBytes: 1,
+      globalInts: 1,
+      globalBytes: 1,
+      payFlags: {}
+    };
+    const appInfo = await executeTransaction(deployer, execParams);
+
+    // should not be stored in checkpoint if in run mode
+    assert.isUndefined(deployer.getSSC("approval.teal", "clear.teal"));
```
[the remark this answers is not in the document]
Comment (amityadav0) [reply packages/algob/test/lib/tx.ts:706]: yes, thanks. tests passed, probably my yarn build is not updated.

File: packages/algob/test/lib/tx.ts [hunk 3]
```diff
@@ -668,3 +668,122 @@ describe("Deploy, Delete transactions test in run mode", () => {
...
+      sign: wtypes.SignType.SecretKey,
+      fromAccount: bobAcc,
+      appID: appInfo?.appID as number,
+      newApprovalProgram: "approval.teal",
+      newClearProgram: "clear.teal",
+      payFlags: {}
+    };
+
+    await executeTransaction(deployer, execParams);
+
```
Comment (ratik21) [inline packages/algob/test/lib/tx.ts:753]:
```suggestion
```

File: packages/algob/test/lib/tx.ts [hunk 4]
```diff
@@ -668,3 +668,122 @@ describe("Deploy, Delete transactions test in run mode", () => {
...
+      approvalProgram: "approval.teal",
+      clearProgram: "clear.teal",
+      localInts: 1,
+      localBytes: 1,
+      globalInts: 1,
+      globalBytes: 1,
+      payFlags: {}
+    };
+    const appInfo = await executeTransaction(deployer, execParams);
+
```
Comment (ratik21) [inline packages/algob/test/lib/tx.ts:771]:
```suggestion
```

**Review by ratik21 — APPROVED** [review]

**Pull request merged**